src/Parser/NoBody.php

```diff
@@ -0,0 +1,19 @@
+<?php
+
+namespace React\Http\Parser;
+
+use Evenement\EventEmitterTrait;
+use React\Http\Request;
+use React\Stream\Util;
+
+class NoBody implements ParserInterface
+{
+    use EventEmitterTrait;
+
+    public function __construct(Request $request)
+    {
+        Util::forwardEvents($request, $this, [
+            'end',
```

Comment:
Same here. NoBody should rather just immediately trigger the 'end' event of ParserInterface instead of the request end, because the connection is then already closed and no need to answer the request :P